### DISCUSSION TITLE
MVVM Community Toolkit

### DISCUSSION THREAD

**Discussion by edmathias** [MarimerLLC/csla#3257] [Questions]
I've been delving into the MVVM Community toolkit that came out almost a year ago.  It appears that CSLA is totally consistent with that framework and should work seamlessly. In fact, many of the MVVM capabilities that it introduces already exist due to CSLA observing the necessary interfaces (as Rocky said in his book/e-books). Are there any advantages in the toolkit in working with CSLA, or trapdoors to be aware of?

**Comment by rockfordlhotka** [comment]
I think the MVVM Community toolkit is the evolution of mvvmlite right? I didn't personally use it, but I know people did use it with WPF, etc.

The one thing I have noticed is that if you use a DI approach to implementing the UI (like Blazor for example), then MVVM frameworks lose a lot of their value, because dependency injection automatically handles a lot of the wiring between view, viewmodel, and model. Things that used to require a framework are now basically built-in with DI.

**Reply by edmathias** [reply]
MVVMLite had basically fallen out of support a few years back and is now deprecated.  James Mantemagno wrote a fairly handy MVVM library that helped until the MVVM Community Toolkit came out, and and he's now recommending his library be abandoned in favor of that.  I never really gave much thought about DI as a value replacement for the various MVVM methodologies, but you are absolutely correct.  I think I had been trying to utilize elements of both approaches in code and it got confusing.  I can now come at it with a new viewpoint.  Thanks!

**Reply by rockfordlhotka** [reply]
I don't mean to imply that MVVM frameworks have no value, because they do!

That said, one of the MVVM framework values _was_ to help wire up the view-viewmodel-model trio - which is now often easier done using the built-in DI capabilities of .NET and CSLA.
